stormevents/nhc/track.py

```diff
@@ -1091,21 +1099,31 @@ def __compute_velocity(data: DataFrame) -> DataFrame:
                 advisory_data.loc[shifted_indices, "latitude"],
             )
 
-            intervals = advisory_data.loc[indices, "datetime"].diff()
-            speeds = distances / (intervals / pandas.to_timedelta(1, "s"))
-            bearings = pandas.Series(inverse_azimuths % 360, index=speeds.index)
-
-            for index in indices:
-                cluster_index = (
-                    advisory_data["datetime"] == advisory_data.loc[index, "datetime"]
+            intervals = (
+                (
+                    abs(
+                        advisory_data.loc[indices, "datetime"].values
+                        - advisory_data.loc[shifted_indices, "datetime"].values
+                    )
                 )
-                advisory_data.loc[cluster_index, "speed"] = speeds[index]
-                advisory_data.loc[cluster_index, "direction"] = bearings[index]
+                .astype("timedelta64[s]")
+                .astype(float)
+            )
+            speeds = pandas.Series(distances / intervals, index=indices)
+            bearings = pandas.Series(inverse_azimuths % 360, index=indices)
+            bearings[pandas.isna(speeds)] = numpy.nan
+            # fill in nans carrying forward, because it is same valid time
+            # and forecast but different isotach.
+            # then fill nans backwards to handle the first time
+            speeds.ffill(inplace=True)
+            bearings.ffill(inplace=True)
```

Comment:
@WPringle where would we get nan to carry forward? We already are getting all indices, we shouldn't run into a case of nan that needs to be filled forward for different isotachs. If possible can you think of an example where this happens?

Reply:
Does `geodetic.inv` return nans when locations are same?

Reply:
For the same forecast time but different isotachs the result will be nan because the tilmestep is zero. So we just want to repeat the result from the first line of the same forecast time.